src/super_context.ml

```diff
@@ -84,26 +324,39 @@ let installed_libs t = t.installed_libs
 let find_scope_by_dir  t dir  = Scope.DB.find_by_dir  t.scopes dir
 let find_scope_by_name t name = Scope.DB.find_by_name t.scopes name
 
-let expand_var_no_root t var = String.Map.find t.vars var
+let expand_vars t ~syntax_version ~var : Var.Kind.t option =
+  begin match String_with_vars.Var.destruct var with
+  | Single _ -> ()
+  | Pair (_, _) ->
+    Exn.code_error "expand_vars can't expand forms"
```

Comment:
I think this error can leak to the user. For instance if they write `%{read:...}` in `(flags ...)` or `(copy_files ...)`.

Reply:
You sure about that? We're making sure to call this on the right map. I.e. the one that contains only forms or only single variables.

Reply:
Can you add a test with a `%{read:...}` inside a `copy_files` just to be sure?